Lib/re/_parser.py

```diff
@@ -776,19 +780,9 @@ def _parse(source, state, verbose, nested, first=False):
                 elif char == "(":
                     # conditional backreference group
                     condname = source.getuntil(")", "group name")
-                    if condname.isidentifier():
-                        condgroup = state.groupdict.get(condname)
-                        if condgroup is None:
-                            msg = "unknown group name %r" % condname
-                            raise source.error(msg, len(condname) + 1)
-                    else:
-                        try:
-                            condgroup = int(condname)
-                            if condgroup < 0:
-                                raise ValueError
-                        except ValueError:
-                            msg = "bad character in group name %r" % condname
-                            raise source.error(msg, len(condname) + 1) from None
+                    if (condname.isdecimal() and condname.isascii() and
+                            (condname[0] != "0" or condname == "0")):
+                        condgroup = int(condname)
                         if not condgroup:
                             raise source.error("bad group number",
```

Comment:
Wouldn't this fail if condname is `"0"`?  If this is intentional I would use a more explicit `condgroup == 0`, otherwise `.isdecimal` already expects at least 1 digit and doesn't allow negative numbers, so condgroup should always be an `int` `>=0`, making the condition always `True`.

Reply:
Yes, it is intentional.

Reply:
Is this because `0` is an invalid group with `(?(...)...|...)`?
The documentation doesn't mention this.

Reply:
It just does not make any sense. Group 0 does not exist until the matching finished.